src/Router/Route.php

```diff
@@ -78,6 +84,7 @@ public function __construct($path, $middleware, $methods = self::HTTP_METHOD_ANY
         $this->path       = $path;
         $this->middleware = $middleware;
         $this->methods    = is_array($methods) ? $this->validateHttpMethods($methods) : $methods;
+        $this->name       = empty($name) ? $path . '^' . implode('_', (array) $this->methods) : $name;
```

Comment:
Underscore is an allowed character in an HTTP method, which could make this problematic. Let's go with `:` or `;` (as neither is allowed in HTTP method names).